ipam/allocator.go

```diff
@@ -867,8 +901,13 @@ func (alloc *Allocator) donateSpace(r address.Range, to mesh.PeerName) {
 		return
 	}
 	alloc.debugln("Giving range", chunk, "to", to)
+	oldRanges := alloc.ring.OwnedAndMergedRanges()
 	alloc.ring.GrantRangeToHost(chunk.Start, chunk.End, to)
 	alloc.persistRing()
+	err := alloc.monitor.HandleUpdate(oldRanges, alloc.ring.OwnedAndMergedRanges())
+	if err != nil {
+		alloc.errorf("HandleUpdate failed: %s", err)
+	}
```

Comment:
There is a rather repetitive pattern in all the HandleUpdate calls, which is:

1. oldRanges := alloc.ring.OwnedAndMergedRanges()
2. manipulate ring
3. newRanges := alloc.ring.OwnedAndMergedRanges()
4. HandleUpdate(oldRanges, newRanges)

So why not have a higher-order function that does all that and takes thunk representing step 2?